### DISCUSSION TITLE
74HC4051 multiplexer wiring / coding assistance

### DISCUSSION THREAD

**Discussion by mbroers** [tttapa/Control-Surface#596] [Q&A] [answered]
I am new to teensy design but really love the idea of this project!  I have plans to do a fader/pot/switch box and started using control-surface with a teensy 3.2.  I successfully got a single fader to work without a multiplexor, then a single fader to work with a multiplexer (74HC4051), but when I added a second fader the two faders seem paired when assigned and don't operate separately.  Im not sure if it is the wiring or the code.  Any assistance is appreciated!

```// Include the library
#include <Control_Surface.h>
 
// Instantiate a MIDI Interface to use
USBMIDI_Interface mixdup;
 
CD74HC4051 mux {
  A0,       // Analog input pin
  {3, 4, 5} // Address pins S0, S1, S2
};
 
// Create an array of potentiometers that send out
// MIDI Control Change messages when you turn the
// potentiometers connected to the eight input pins of
// the multiplexer
CCPotentiometer volumePotentiometers[] {
  { mux.pin(0), { MIDI_CC::Channel_Volume, CHANNEL_1 } },
  { mux.pin(1), { MIDI_CC::Channel_Volume, CHANNEL_2 } }
};

// Initialize the Control Surface
void setup() {
  Control_Surface.begin();
}
 
// Update the Control Surface
void loop() {
  Control_Surface.loop();
}
```
![IMG_4218](https://user-images.githubusercontent.com/2566312/138359816-78db875e-f53e-4cc3-b5dd-b8dd19f01cad.jpg)

**Comment by tttapa** [comment]
The pin numbering on the Teensy looks incorrect, please see https://www.pjrc.com/store/teensy32.html#pins. The pin numbers in the code are the digital pin numbers, not the physical package pin numbers.

**Comment by mbroers** [comment]
Thanks for clearing it up - that makes sense!  I'll report back when I have a chance to test it.

**Comment by mbroers** [comment]
I've made the changes, directed S0, S1, S2 to _digital_ pins 3, 4, and 5 based on the teensy pin out diagram and thought I read somewhere to use the 3.3v power pin of the teensy for the 4051 vcc so I've done that as well. I still have the same problem with the two pots that are wired in parallel effecting each other. It seems like I dont have the pots wired up correctly.. Im going to keep futzing but it is a bit odd.

**Comment by mbroers** [comment]
Ive checked the voltage at the multiplexer pins when I adjust each pot separately, the voltage behavior seems acceptable:

If I am testing the range of A0 on the multiplexer the voltage measure there goes from 0~3.3v and the voltage measured on A1 remains mostly unaffected, but with a small bump to .1v at the extreme end of the A0 pot (I think this is fine?).  

It seems like somehow my s0/s1/s2 are being pulled to ground even with all the ICs removed, so I think this is faulty wiring of a messed up breadboard...

**Comment by mbroers** [comment, marked as answer]
Ok so the issue was I didnt realize the breadboard I was working with had bus layers. I re-implemented on a fresh vanilla breadboard and am getting the results I expect! Sorry to open a dumb discussion but it definitely helped me get the digital pins routed correctly. Thanks again!!
